### ISSUE TITLE
SalesforceAuthenticationFailed: invalid_grant: expired authorization code

### ISSUE THREAD

**Issue by VdeJong** [simple-salesforce/simple-salesforce#491]
I'm getting this issue using this package and JWT Token authentication.

```
Salesforce(username=self.username,
                   consumer_key=self.issuer,
                   privatekey=self.get_key_file(),
                   domain=self.domain)
```

It works when I use the requests package:
```
        private_key = self.get_key_file()

        claim = {
            "iss": self.issuer,
            "exp": self.expiration_time,
            "aud": self._base_url,
            "sub": self.username,
        }
        assertion = jwt.encode(
            claim, private_key, algorithm="RS256", headers={"alg": "RS256"}
        )

        response = requests.post(
            f"{self._base_url}/services/oauth2/token",
            data={
                "grant_type": "urn:ietf:params:oauth:grant-type:jwt-bearer",
                "assertion": assertion,
            },
        )
```

I found out it's because of the expiration time that is calculated wrong. In my own code example:
`self.expiration_time = int(time.time()) + 300  # 5 minutes`

in the package:
`expiration = datetime.utcnow() + timedelta(minutes=3)`
`expiration_time = '{exp:.0f}'.format(exp=time.mktime(expiration.timetuple()) + expiration.microsecond / 1e6)`

If that is changed to `int(time.time()) + 300` or `datetime.now() + timedelta(minutes=3)` it works, because `datetime.utcnow()` doesn't seem to work for me as I'm not in that timezone.

**Comment by sandello-alkr** [comment]
Same issue here, could the mentioned PR #431 be merged any time soon? It's a deal breaker at the moment.

**Comment by jon-wobken** [comment]
@sandello-alkr - I'll review and merge this weekend and release a dot release for PyPi.

**Comment by sandello-alkr** [comment]
@jon-wobken any update on this?

**Comment by cbailiss** [comment]
This is also a blocker for us.

**Comment by jon-wobken** [comment]
@sandello-alkr & @cbailiss  - I just merged #431  this morning. It should resolve this so closing this out.
